### ISSUE TITLE
Problems when debugging Brackets

### ISSUE THREAD

**Issue by rioc0719** [adobe/brackets#10315]
When I'm using Brackets' devtools (accessed from the Debug menu), I cannot select elements, collapse a tree, and a tree only expands with a double click. I traced the problem down to a function in inspector.js (localhost:9234/devtools/inspector.js) called `TreeElement.prototype.isEventWithinDisclosureTriangle` wherein someone uses `getPropertyCssValue` instead of `getPropertyValue`. [See here] (https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration) for the appropriate docs. I'm using release 1.1 ("ea908cae5"). This has not worked since I started using Brackets at version 0.43, and I'm on Ubuntu Gnome 14.10.
I know how to fix this, but I can't find the correct file in the source code. If someone could point me in the right direction, I know how to go from there.

**Comment by redmunds** [comment]
@rioc0719 Chrome's *built-in* devtools is for your document. Navigating to localhost:9234 gives you the devtools for Brackets itself.

**Comment by rioc0719** [comment]
@redmunds So localhost:9234 uses Chrome's built-in devtools to debug Brackets, not some local instance of the devtools?

**Comment by redmunds** [comment]
> So localhost:9234 uses Chrome's built-in devtools to debug Brackets, not some local instance of the devtools?

Yes.

**Comment by peterflynn** [comment]
Hmm, I think there might still be some confusion here. To clarify:

* When you choose _Debug > Show Developer Tools_, a window opens showing dev tools for Brackets itself (you can never use it to debug your own project). This copy of dev tools is built into Brackets.
* When you go to localhost:9234 in your browser, you see _the same thing_ -- dev tools for Brackets itself.  You are still using a copy that is built into Brackets -- _not_ your Chrome browser's version of dev tools.  (As far as I know, there's no way to point Chrome's copy of dev tools at Brackets).
    * Also note -- this is only supported in Chrome; it may not work if you open localhost:9234 in another browser such as FF or Safari.
* The only way to open dev tools for your own project is the usual way you'd do it without Brackets running.  This will always be Chrome's copy of dev tools; it is never the version built into Brackets.

@rioc0719 I'm wondering if you could clarify the original bug report. When you say "I cannot select elements, collapse a tree, and a tree only expands with a double click," are you talking about the tree in the dev tools UI (in the Sources view?), or the file tree in the Brackets UI?  Also, can you confirm that dev tools is opening in a Brackets window for you (not in a browser tab)?

One thing you can try is [clearing your Brackets cache](https://github.com/adobe/brackets/wiki/Cache-Folder) -- that should basically be a 'factory reset' of the built-in copy of dev tools (at least as long as you're opening it in Brackets and not in a browser tab).

**Comment by rioc0719** [comment]
@peterflynn Thank you for the clarification. When I'm talking about the trees, I'm referring to the ones in the Elements panel in the devtools, and those in the Sources panel. Also, the devtools are opening in a browser tab in Chrome.

**Comment by rioc0719** [comment]
@peterflynn Also, clearing the cache does not fix the problem.

**Comment by peterflynn** [comment]
Ohh I forgot Linux doesn't have the Brackets 1.1 shell yet -- so _Debug > Show Developer Tools_ does indeed open a tab in Chrome (instead of a Brackets window) on the Linux version, for now.  What version of Chrome are you running?

**Comment by rioc0719** [comment]
Chrome unstable 41.0.2251.0, 64-bit.

**Comment by rioc0719** [comment]
Upon further investigation, it seems that a lot of the code that creates the devtools is outdated. It uses the  obsolete `CSSPrimitiveValue` interface, which is still available in the CEF, but not in Chrome.

**Comment by peterflynn** [comment]
Yes, the 1.0 Brackets shell is using a pretty old version of Chrome.  The 1.1 shell is much newer but we haven't been able to get it working for Linux yet -- see adobe/brackets-shell#489.  Once that's fixed, this problem should go away.

In the meantime, I think opening the URL in stable Chrome instead of unstable should work...

**Comment by rioc0719** [comment]
@peterflynn Okay, thank you.

**Comment by peterflynn** [comment]
@rioc0719 Does the test build at https://github.com/adobe/brackets/releases/tag/linux-cef-2171 work for you?  This build should open dev tools in a window within Brackets by default, but you should also be able to open a Chrome tab and manually go to localhost:9234 to see the same dev tools.  Do both of those work for you with your unstable Chrome 41?

**Comment by rioc0719** [comment]
@peterflynn The dev tools in a Brackets window work correctly, however localhost:9234 still has the same problem.

**Comment by peterflynn** [comment]
I guess `CSSPrimitiveValue` was removed _super_ recently then -- the dev tools in that build of Brackets are from Chrome 39, so they must have made coordinated changes in both Chrome & the dev tools code in a short timespan since then.

This could be a problem though, since the localhost:9234 option is important for debugging cases where Brackets is too hosed to even boot up the menu bar...

**Comment by rioc0719** [comment]
@peterflynn `CSSPrimitiveValue` isn't available in Chrome 40 (the latest stable).

**Comment by peterflynn** [comment]
Ah, didn't notice Chrome 40 just went stable today...

Looks like dev tools was just updated at the end of October (part of Chrome 40): https://chromium.googlesource.com/chromium/blink/+/8dcc811b8f68c0537c7b451359f4ac0097c1b13c%5E%21/.  Our CEF is slightly older (Chrome 39), so that part of dev tools will be broken until our _next_ CEF update after the one we're just wrapping up now.  Could be several months, since updating CEF is often a huge pain.

**Comment by peterflynn** [comment]
@rioc0719 As a workaround note that you can open any file by pressing Ctrl-O and then typing (part of) the filename.  That hopefully means you don't need to click in the dev tools tree widget at all.

**Comment by rioc0719** [comment]
@peterflynn Since the devtools in a Brackets window works correctly with no errors, I could just use that.

**Comment by rioc0719** [comment]
Well, actually, the devtools I'm talking about have non-stop WebSocket errors, but it's not affecting their usability.

**Comment by peterflynn** [comment]
Yep, that too -- as long as you never get Brackets in _too_ broken of a state ;-)

**Comment by rioc0719** [comment]
@peterflynn This build is working pretty well, I think I'll just stick with it for now. The only problem is that CSSLint won't update, but that's nothing, really.

**Comment by peterflynn** [comment]
Generalizing title since this seems to affect all expand/collapse widgets in the dev tools UI.  Tempted to add 'Linux only,' but this actually affects all OSes in the case where Brackets is hosed badly enough that the _Debug_ menu isn't functioning.
